### PR TITLE
Updates to crypto_test app; STM32 crypto HW driver; CBC/CTR speed

### PR DESCRIPTION
**Pull request by utzig**
* apps: crypto_test: add benchmarks for CBC/CTR
* stm32: crypto: make driver obey syscfg HW support
* crypto: use 32-bit XOR in CBC/CTR

### CODE REVIEW

**Comment by utzig** [comment]
Could someone take a look?

File: hw/drivers/crypto/src/crypto.c [hunk 1]
```diff
@@ -56,8 +61,20 @@ crypto_do_ctr(struct crypto_dev *crypto, const void *key, uint16_t keylen,
...
-            outbuf8[i] = inbuf8[i] ^ _out[i];
+        /*
+         * For full blocks increase speed by doing 32-bit XOR; maintain the
+         * stream semantics doing byte XORs for smaller sizes (end of buffer).
+         */
+        if (len == AES_BLOCK_LEN) {
+            inbuf32 = (uint32_t *)inbuf8;
+            outbuf32 = (uint32_t *)outbuf8;
+            for (i = 0; i < len / 4; i++) {
+                outbuf32[i] = inbuf32[i] ^ _out32[i];
```
Comment (kasjer) [inline hw/drivers/crypto/src/crypto.c]: I'm not sure what is the usage of this function but unaligned access here could crash Cortex-M0

File: apps/crypto_test/src/main.c [hunk 2]
```diff
@@ -282,6 +286,66 @@ run_benchmark(char *name, block_encrypt_func_t encfn, void *data, uint8_t iter)
...
+                printf("fail... blkidx=%u\n", blkidx);
+                for (j = 0; j < AES_BLOCK_LEN; j++) {
+                    printf("[%02x]<%02x> ", output[j],
+                            aes_128_cbc_expected[blkidx + j]);
+                }
+                return;
+            }
+        }
+    }
+    printf("done in %lu ticks\n", os_time_get() - t);
```
Comment (kasjer) [inline apps/crypto_test/src/main.c]: I would not use ticks to display time, you could not compare results of two different boards not knowing what tick mean on each one.

File: hw/drivers/crypto/src/crypto.c (hunk 1 above)
Comment (utzig) [reply hw/drivers/crypto/src/crypto.c]: This should be fixed now, thanks!

File: apps/crypto_test/src/main.c (hunk 2 above)
Comment (utzig) [reply apps/crypto_test/src/main.c]: Comparing to different boards is not relevant IMO, what is being done here is to compare the HW acceleration speed to SW implementations using tinycrypt/mbedTLS.

**Pull request merged**